pkg/redis/redis_client.go

```diff
@@ -166,8 +186,29 @@ func (c *redisClient) preventOOMByBackoff(wrappedCmd func() (interface{}, error)
 		return err
 	}
 
-	// No further error handling as the wrapped redis error is handled elsewhere
 	err = backoff.RetryNotify(operation, backoffConfig, notify)
 
-	return res
+	return res, err
+}
+
+func (c *redisClient) waitForConnection() bool {
+	backoffConfig := backoff.NewExponentialBackOff()
```

Comment:
We should use the same backof config as for the OOM case, to have it in the same place (and configurable eventually). The important one here is also `MaxElapsedTime = 0`, otherwise the client does not completely block